src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java

```diff
@@ -52,139 +45,185 @@
-                // Creating tabix indices with a non standard extensions can cause problems so we disable it
-                if (!indexFile.getAbsolutePath().endsWith(TabixUtils.STANDARD_INDEX_EXTENSION)) {
-                    throw new UserException("The index for " + featureFile + " must be written to a file with a \"" + TabixUtils.STANDARD_INDEX_EXTENSION + "\" extension");
-                }
+    private void writeVCFHeader(VariantContextWriter vcfWriter) {
+        // setup the header fields
+        final VCFHeader inputHeader = getHeaderForVariants();
+        final Set<VCFHeaderLine> inputHeaders = inputHeader.getMetaDataInSortedOrder();
+        final Set<VCFHeaderLine> hInfo = new HashSet<>(inputHeaders);
+
+        if( tranches.size() >= 2 ) {
+            for(int i = 0; i < tranches.size() - 1; i++) {
+                String filterKey = filterKeyFromTranches(infoKey, tranches.get(i), tranches.get(i+1));
+                String filterDescription = filterDescriptionFromTranches(infoKey, tranches.get(i), tranches.get(i+1));
+                hInfo.add(new VCFFilterHeaderLine(filterKey, filterDescription));
+            }
+        }
+        String filterKey = filterKeyFromTranches(infoKey, tranches.get(tranches.size()-1), 100.0);
+        String filterDescription = filterDescriptionFromTranches(infoKey, tranches.get(tranches.size()-1), 100.0);
+        hInfo.add(new VCFFilterHeaderLine(filterKey, filterDescription));
+        final TreeSet<String> samples = new TreeSet<>();
+        samples.addAll(inputHeader.getGenotypeSamples());
+        hInfo.addAll(getDefaultToolVCFHeaderLines());
+        final VCFHeader vcfHeader = new VCFHeader(hInfo, samples);
+        vcfWriter.writeHeader(vcfHeader);
+    }
+
+    private String filterKeyFromTranches(String infoKey, double t1, double t2){
+        return String.format("%s_Tranche_%.2f_%.2f", infoKey, t1, t2);
+    }
 
-                return IndexFactory.createIndex(featureFile, codec, IndexFactory.IndexType.TABIX, null);
+    private String filterDescriptionFromTranches(String infoKey, double t1, double t2){
+        return String.format("Truth sensitivity between %.2f and %.2f for info key %s", t1, t2, infoKey);
+    }
+
+    private boolean isTrancheFiltered(double score, List<Double> cutoffs) {
+        return score <= cutoffs.get(0);
+    }
 
-            } else {
-                // Optimize indices for other kinds of files for seek time / querying
-                return IndexFactory.createDynamicIndex(featureFile, codec, IndexFactory.IndexBalanceApproach.FOR_SEEK_TIME);
+    private String filterStringFromScore(double score, List<Double> cutoffs){
+        for (int i = 0; i < cutoffs.size()-1; i++){
+            if (score > cutoffs.get(i) && i == 0){
+                throw new GATKException("Trying to add a filter to a passing variant.");
+            } else if (score > cutoffs.get(i)){
+                return filterKeyFromTranches(infoKey, tranches.get(i), tranches.get(i+1));
```

Comment:
Is this assigning the correct tranche label ? `i` is always > 0 && < size if you get here, should the tranche label for this be `i-1` to `i` ? `i+1` would go off the end of the list when `i == size()-1` ?

Reply:
I think this is OK.  Tranches and cutoffs are assumed to be the same size and loop conditional is 
`i < cutoffs.size()-1`

Reply:
@lucidtronix Right, I just want to verify that the label assignment is correct: Given these tranches and snp cutoffs:

Tranche: 95.0
Tranche: 99.0
Tranche: 99.9

snpCutoffs: 0.860204
snpCutoffs: -1.59729
snpCutoffs: -4.4343

and a snpScore of -3.51172, this assigns filter label 99.90 - 100.00.

Reply:
Oh I see now, you're right that is not expected. I will shift the tranches over.